### PR TITLE
[ESIMD][doc] Added documentation for some ESIMD math APIs

### PR DESCRIPTION
**Pull request by DenisBakhvalov**
Part 1.

Documented the following APIs:
  sat, abs,
  shl, shr, rol, ror, asr, lsr
  qout, mod, div
  min, max
  dp2, dp3, dp4, dph, line
  frc

### CODE REVIEW

File: sycl/include/sycl/ext/intel/experimental/esimd/math.hpp [hunk 1]
```diff
@@ -75,6 +82,14 @@ __esimd_abs_common_internal(T1 src0, int flag = saturation_off) {
 }
 } // namespace detail
 
+/// Get absolute value (vector version)
+/// \tparam T0 element type of the returned vector.
+/// \tparam T1 element type of the input vector.
+/// \tparam SZ size of the input and returned vector.
+/// @param src0 the input vector.
+/// @param flag saturation on or off(default). 0 - no saturation, 1 - perform
+/// operation with saturation.
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/math.hpp:91]:
```suggestion
/// @param flag saturation on or off(default). saturation_on/saturation_off value enables/disables the
/// saturation.
```

File: sycl/include/sycl/ext/intel/experimental/esimd/math.hpp [hunk 2]
```diff
@@ -198,14 +273,29 @@ esimd_shr(T1 src0, T2 src1, int flag = saturation_off) {
...
+/// the input vector \p src0 shall be rotated.
+/// @return vector of rotated elements.
 template <typename T0, typename T1, int SZ>
 ESIMD_NODEBUG ESIMD_INLINE typename sycl::detail::enable_if_t<
     std::is_integral<T0>::value && std::is_integral<T1>::value, simd<T0, SZ>>
 esimd_rol(simd<T1, SZ> src0, simd<T1, SZ> src1) {
   return __esimd_rol<T0, T1, SZ>(src0, src1);
 }
 
+/// Rotate left operation (vector version)
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/math.hpp:291]: (here and if few more places) Description matches the previous function, even though they are different. Please update.

File: sycl/include/sycl/ext/intel/experimental/esimd/math.hpp [hunk 3]
```diff
@@ -801,6 +1173,16 @@ ESIMD_NODEBUG ESIMD_INLINE
   return Result;
 }
 
+/// FIXME: linear equation.
```
Comment (kbobrovs) [inline sycl/include/sycl/ext/intel/experimental/esimd/math.hpp:1176]: Is there a difference with line 989?

File: sycl/include/sycl/ext/intel/experimental/esimd/math.hpp (hunk 1 above)
Comment (DenisBakhvalov) [reply sycl/include/sycl/ext/intel/experimental/esimd/math.hpp]: Done.

File: sycl/include/sycl/ext/intel/experimental/esimd/math.hpp (hunk 2 above)
Comment (DenisBakhvalov) [reply sycl/include/sycl/ext/intel/experimental/esimd/math.hpp]: Done

File: sycl/include/sycl/ext/intel/experimental/esimd/math.hpp (hunk 3 above)
Comment (DenisBakhvalov) [reply sycl/include/sycl/ext/intel/experimental/esimd/math.hpp:1177]: Not yet, but it's a `FIXME`.

**Pull request merged**